vowpalwabbit/interactions.h

```diff
@@ -74,8 +77,16 @@ std::vector<std::vector<namespace_index>> compile_interaction(
   }
 
   // Quadratic fast path or generic generation function.
-  auto result = num_wildcards == 2 ? expand_quadratics_wildcard_interactions(leave_duplicate_interactions, indices)
-                                   : generate_func(indices, num_wildcards);
+  auto result_indices = num_wildcards == 2
+      ? expand_quadratics_wildcard_interactions(leave_duplicate_interactions, indices)
+      : generate_func(indices, num_wildcards);
+  std::vector<std::vector<INTERACTIONS::interaction_term>> result;
```

Comment:
this is on the hot path right? it might be worth seeing if we can have the `result` vector be a member var who's size gets stabilized over runs and see if that affects the benchmarks

Reply:
I wouldn't say this is on the hot path. It's only run when the set of seen namespaces change

Reply:
should we be concerned that some of the benchmarks are a bit red?

Reply:
Hmm, some of them are fine and some are bit red yeah. The ones that were added recently seem to have more variance I've noticed, so I am less concerned about those.

In general, I did expect this PR to have a slight negative effect on performance. Given this is adding a branch into the interactions stuff it seems reasonable. One possible way is to keep both "types" of interactions completely separate and then invoke the interactions loop twice - one for each type. This introduces more complexity elsewhere so I don't know if it would pay off. I haven't implemented it since it will require a fair few changes.

Reply:
As long as we are aware of where the regressions come from, so that in the end we can focus on those first to optimize should the final solution be too slow, then that's fine by me

Reply:
Yeah I 100% agree